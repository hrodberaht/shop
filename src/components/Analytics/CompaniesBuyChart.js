import React, { Component } from 'react';
import Highcharts from 'highcharts';
import Exporting from 'highcharts/modules/exporting';

Exporting(Highcharts);

export default class CompaniesBuyChart extends Component {
  componentDidMount() {
    Highcharts.chart('companies-buy-chart', {
      chart: {
        type: 'bar',
      },
      title: {
        text: 'Sells',
      },
      xAxis: {
        categories: ['Sumary'],
      },
      yAxis: {
        title: {
          text: 'Sells',
        },
      },
      series: [
        {
          name: 'IBM',
          data: [2400],
        },
        {
          name: 'Apple',
          data: [4800],
        },
      ],
    });
  }

  render() {
    return <div id="companies-buy-chart" />;
  }
}
