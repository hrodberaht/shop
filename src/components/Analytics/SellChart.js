import React, { Component } from 'react';

export default class SellChart extends Component {
  render() {
    return <div>SellCharts</div>;
  }
}
