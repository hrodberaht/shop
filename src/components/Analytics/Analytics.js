import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import SellChart from './SellChart';
import CompaniesBuyChart from './CompaniesBuyChart';

import { fetchSoldProducts } from '../../store/charts/actionCreators';
import { getChartsDataSoldProducts } from '../../store/charts/selectors';
import { getAuthUserId } from '../../store/authenticate/selectors';

export class Analytics extends Component {
  componentDidMount() {
    const { userId, soldProducts } = this.props;
    soldProducts(userId);
  }

  render() {
    const { dataSellChart, userId } = this.props;
    return (
      <div>
        {dataSellChart.length === 0 ? <p>Loading...</p> : <SellChart data={dataSellChart} />}
        <Query
          query={gql`
            {
              orders {
                totalPrice
                companyId
              }
            }
          `}
          fetchPolicy="network-only"
          variables={{ id: userId }}
        >
          {data => <CompaniesBuyChart fetchedData={data} />}
        </Query>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  dataSellChart: getChartsDataSoldProducts(state),
  userId: getAuthUserId(state),
});

export default connect(
  mapStateToProps,
  { soldProducts: fetchSoldProducts },
)(Analytics);

Analytics.propTypes = {
  dataSellChart: PropTypes.arrayOf(PropTypes.object).isRequired,
  soldProducts: PropTypes.func.isRequired,
  userId: PropTypes.string.isRequired,
};
