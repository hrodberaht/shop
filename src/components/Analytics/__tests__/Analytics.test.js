import { Analytics } from '../Analytics';

const defaultProps = {
  soldProducts: jest.fn(),
  userId: '1234',
  dataSellChart: [],
};

const setup = buildSetup(Analytics, defaultProps);

describe('<Analytics />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
