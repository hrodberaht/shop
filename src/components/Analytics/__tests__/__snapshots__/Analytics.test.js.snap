// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Analytics /> should render without crash 1`] = `
<div>
  <p>
    Loading...
  </p>
  <Query
    fetchPolicy="network-only"
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": undefined,
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "orders",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "totalPrice",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "companyId",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 131,
          "start": 0,
        },
      }
    }
  >
    <Component />
  </Query>
</div>
`;

exports[`<Analytics /> should render without crash if retrive dataSellChart  1`] = `
<div>
  <SellChart
    data={
      Array [
        Object {},
      ]
    }
  />
  <Query
    fetchPolicy="network-only"
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": undefined,
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "orders",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "totalPrice",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "companyId",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 131,
          "start": 0,
        },
      }
    }
  >
    <Component />
  </Query>
</div>
`;
